presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQueryFactory.java

```diff
@@ -124,7 +123,7 @@ public DispatchQuery createDispatchQuery(
             }
             catch (Throwable e) {
                 stateMachine.transitionToFailed(e);
-                queryMonitor.queryImmediateFailureEvent(stateMachine.getBasicQueryInfo(Optional.empty()), toFailure(e));
+                // queryMonitor.queryImmediateFailureEvent will be called by state change listenere registered in DispatchManager.queryCreated
```

Comment:
Remove.
Instead, please add a test that would fail if this code is reinserted here.

Reply:
I will try. May be hard